blog/2025-08-00-uppy-5.0.mdx

```diff
@@ -0,0 +1,772 @@
+
+Common icons, such as a webcam, device, Dropbox, Google Drive, OneDrive, etc.
+
+**Props**:
+
+- `provider`:
+  `'device' | 'camera' | 'screen-capture' | 'audio' | 'dropbox' | 'facebook' | 'instagram' | 'onedrive' | 'googlephotos' | 'googledrive'`
+- `fill?: string`
+
+<ProviderIconDemo />
+
+  </TabItem>
+
+</Tabs>
+
+#### Styling component states
+
+Components expose their internal state through data attributes, enabling precise
+CSS targeting.
+
+```css
+button[data-uppy-element='upload-button'][data-state='init'] {
+	background-color: #3b82f6;
+}
+
+button[data-uppy-element='upload-button'][data-state='uploading'] {
+	background-color: #f59e0b;
+	cursor: wait;
+}
+
+button[data-uppy-element='upload-button'][data-state='complete'] {
+	background-color: #10b981;
+}
+
+button[data-uppy-element='upload-button'][data-state='error'] {
+	background-color: #ef4444;
+}
+```
+
+### New hooks
+
+Examples are shown in React, but the same hooks are available in Vue and Svelte.
+
+<Tabs>
+  <TabItem value="use-dropzone" label="useDropzone">
+
+Here is an example that uses hooks to exactly mimic the `<Dropzone />` component.
+
+```tsx
+import clsx from 'clsx';
+import { useDropzone } from '@uppy/react';
+
+type DropzoneProps = {
+	note?: string;
+	noClick?: boolean;
+};
+
+export function Dropzone({ note, noClick }: DropzoneProps) {
+	const { getRootProps, getInputProps } = useDropzone({ noClick });
+
+	return (
+		<div data-uppy-element="dropzone" role="presentation">
+			<input
+				{...getInputProps()}
+				tabIndex={-1}
+				name="uppy-dropzone-file-input"
+				className="hidden"
+			/>
+			<div
+				{...getRootProps()}
+				tabIndex={0}
+				className={clsx(
+					'border-2 border-dashed border-gray-300',
+					'rounded-lg p-6 bg-gray-50',
+					'transition-colors duration-200',
+					{
+						'cursor-pointer hover:bg-blue-50': !noClick,
+					},
+				)}
+			>
+				<div className="flex flex-col items-center justify-center h-full space-y-3">
+					<p className="text-gray-600">Drop files here or click to add them</p>
+				</div>
+				{note ?
+					<div className="text-sm text-gray-500">{note}</div>
+				:	null}
+			</div>
+		</div>
+	);
+}
+```
+
+  </TabItem>
+
+  <TabItem value="use-file-input" label="useFileInput">
+
+Create a file input button that opens a file picker when clicked and add the
+selected files to Uppy.
+
+```tsx
+import { useFileInput } from '@uppy/react';
+
+type FileInputProps = {
+	multiple?: boolean;
+	accept?: string;
+};
+
+export function FileInput({ multiple, accept }: FileInputProps) {
+	const { getButtonProps, getInputProps } = useFileInput({ multiple, accept });
+
+	return (
+		<div>
+			<input {...getInputProps()} className="hidden" />
+			<button {...getButtonProps()}>Add files</button>
+		</div>
+	);
+}
+```
+
+  </TabItem>
+
+  <TabItem value="use-remote-source" label="useRemoteSource">
+
+This hook can be used to build a UI for remote sources, such as Dropbox or Google
```

Comment:
As an example to underline the previous comment about adding headers: it here simply says 'this hook'. Because the structure is not crystal clear, it can be confusing whether this line says something about the code above, or about the code below.

Of course, you can find out by reading the code, but it seems much better to have this cleared up to begin with.